src/components/SelectedAnnotation.jsx

```diff
@@ -60,10 +64,43 @@ class SelectedAnnotation extends React.Component {
     });
   }
 
+  onKeyUp(e) {
+    const character = Utility.getHebrewChar(e);
+
+    if (character !== false) {
+      this.props.dispatch(pressedKey(null));
+    }
+  }
+
+  onKeyDown(e) {
+    const character = Utility.getHebrewChar(e);
+
+    if (Utility.getKeyCode(e) === KEY_VALUES.Enter) {
+      this.saveText();
+    }
+
+    if (character !== false) {
+      e.preventDefault();
+      this.setState({
+        annotationText: this.inputText.value + character.unicode
```

Comment:
Does this add new text to the end of the existing text, even if the text carat is in the middle of the text in the textarea? For example, if I've clicked on a word so I can correct a spelling mistake.

Reply:
Ahh, good call. This is something I need to straighten out. I will have to put the `<input>` into more of a test drive.

Reply:
After working on the text highlighter tools for PFE, working with the [Selection API](https://developer.mozilla.org/en-US/docs/Web/API/Selection) can get quite complicated too. I wonder if there's a way to let people edit Latin text in the input, just like they would normally, then translate the characters to Hebrew on the fly?

Reply:
What do you mean by translate the characters to Hebrew on the fly? Users are expected to be typing in Hebrew characters directly from the script without the need for translation.

Reply:
I believe what Jim means by "on the fly" is something like this...
```
textInput.onChange = () => {  //Whenever the user types in a new character...
  textInput.text = textInput.text  //...do an "autocorrect" for each letter.
  .replace(/a/ig, HEBREW.ALEF)
  .replace(/b/ig, HEBREW.BETH)
  .replace(/g/ig, HEBREW.GIMMEL)
  ...;
}
```

There are pros and cons to this method, namely:
1. it respects input position (e.g. if the user moves the cursor to the middle of the word and types a character)
2. users can never type or copy-paste English characters. (Wait is this a pro or a con?)
3. it _doooesn't_ (?) interfere with copy-pasting of Hebrew text.
4. it _shooould_ (?) work with native Hebrew keyboards.
5. we don't need to capture keyboard input, which makes it simpler to implement (i.e. no need to have listeners) and avoids us accidentally capturing keyboard shortcuts.

Reply:
Yes. Sorry for the slow reply. I thought it would maybe avoid having to work with browser text selection APIs if you had the volunteers edit Latin text, from a Latin keyboard, and just translated the alphabet to Hebrew when you display the text. I am not sure how feasible this is, but it is easier to implement than a full text editor in JavaScript.

Reply:
I hadn't even thought about trying to detect cut-and-paste events. I believe [React's synthetic event](https://reactjs.org/docs/events.html) supports `onPaste` but I've never actually tried it to see how it works in different browsers.

Reply:
Ok. This makes sense and might be the approach I wind up taking. I have started today looking at the Selection API, but it is proving to be a bit tricky. Also, using the `onChange` function rather than `onKeyDown` will help with some of the input changes.

Reply:
This makes sense but I'm having the issue of the cursor moving to the end of the input if the `this.inputText.value` is manipulated with `onChange`, which can be an issue when adding for than one letter in the middle of a word.